### PR TITLE
Implemented SmeltableProperty

### PR DESCRIPTION
**Pull request by Redrield**
Implementation for [Sponge API #1325](https://github.com/SpongePowered/SpongeAPI/pull/1325)

### CODE REVIEW

File: src/main/java/org/spongepowered/common/data/property/store/item/SmeltablePropertyStore.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import net.minecraft.item.crafting.FurnaceRecipes;
+import org.spongepowered.api.data.property.item.SmeltableProperty;
+import org.spongepowered.common.data.property.store.common.AbstractItemStackPropertyStore;
+
+import java.util.Optional;
+
+public class SmeltablePropertyStore extends AbstractItemStackPropertyStore<SmeltableProperty> {
+
+    @Override protected Optional<SmeltableProperty> getFor(ItemStack itemStack) {
+        if(FurnaceRecipes.instance().getSmeltingResult(itemStack) != null) {
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/data/property/store/item/SmeltablePropertyStore.java]: `if (`

File: src/main/java/org/spongepowered/common/data/property/store/item/SmeltablePropertyStore.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+import net.minecraft.item.ItemStack;
+import net.minecraft.item.crafting.FurnaceRecipes;
+import org.spongepowered.api.data.property.item.SmeltableProperty;
+import org.spongepowered.common.data.property.store.common.AbstractItemStackPropertyStore;
+
+import java.util.Optional;
+
+public class SmeltablePropertyStore extends AbstractItemStackPropertyStore<SmeltableProperty> {
+
+    @Override protected Optional<SmeltableProperty> getFor(ItemStack itemStack) {
```
Comment (kashike) [inline src/main/java/org/spongepowered/common/data/property/store/item/SmeltablePropertyStore.java]: Annotation on it's own line.

File: src/main/java/org/spongepowered/common/data/property/store/item/SmeltablePropertyStore.java (hunk 1 above)
Comment (liach) [inline src/main/java/org/spongepowered/common/data/property/store/item/SmeltablePropertyStore.java]:
```java
return Optional.of(new SmeltableProperty(FurnaceRecipes.instance().getSmeltingResult(itemStack) != null));
```

**Comment by Aaron1011** [comment]
@Redrield: Can you replace this against `bleeding`?

**Pull request closed without merging**